In [1]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [9]:
df = pd.read_csv('/mnt/c/Users/marcn/Documents/Ironhack/sharky/Data/attacks.csv', encoding='cp1252')
print(df.shape)
display(df.info())
df.sample(3)

(25723, 24)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25723 entries, 0 to 25722
Data columns (total 24 columns):
 #   Column                  Non-Null Count  Dtype  
---  ------                  --------------  -----  
 0   Case Number             8702 non-null   object 
 1   Date                    6302 non-null   object 
 2   Year                    6300 non-null   float64
 3   Type                    6298 non-null   object 
 4   Country                 6252 non-null   object 
 5   Area                    5847 non-null   object 
 6   Location                5762 non-null   object 
 7   Activity                5758 non-null   object 
 8   Name                    6092 non-null   object 
 9   Sex                     5737 non-null   object 
 10  Age                     3471 non-null   object 
 11  Injury                  6274 non-null   object 
 12  Fatal (Y/N)             5763 non-null   object 
 13  Time                    2948 non-null   object 
 14  Species                 34

None

Case Number Date  Year Type Country Area Location Activity Name Sex   \
11985         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
20320         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   
21564         NaN  NaN   NaN  NaN     NaN  NaN      NaN      NaN  NaN  NaN   

       ... Species  Investigator or Source  pdf href formula href  \
11985  ...      NaN                    NaN  NaN          NaN  NaN   
20320  ...      NaN                    NaN  NaN          NaN  NaN   
21564  ...      NaN                    NaN  NaN          NaN  NaN   

      Case Number.1 Case Number.2 original order Unnamed: 22 Unnamed: 23  
11985           NaN           NaN            NaN         NaN         NaN  
20320           NaN           NaN            NaN         NaN         NaN  
21564           NaN           NaN            NaN         NaN         NaN  

[3 rows x 24 columns]

We have 23 parameters related to 25723 different cases.
The good news is that we have many cases, the bad news is that the data has a lot of NaNs to be filled. 

We need to filter the data in order to work with it. Decide wether to keep or not each column. 
